File: src/falcon_core/math/labelled_arrays/labelled_control_arrays.py
"""A group of doamin arrays that have been labelled and kept together."""

from .base_labelled_arrays import BaseLabelledArrays
from .dependancies import TypeVar, arrayND
from .labelled_control_array import LabelledControlArray

U = TypeVar("U", bound=arrayND)


class LabelledControlArrays[U: arrayND](BaseLabelledArrays[LabelledControlArray[U]]):
    """A series of domain arrays that have been labelled and kept together."""

    def __init__(self, arrays: list[LabelledControlArray[U]]) -> None:
        """Initialize the LabelledArrays object."""
        super().__init__(arrays=arrays)

File: src/math/arrays/LabelledControlArrays.cpp
#include "falcon_core/math/arrays/LabelledControlArrays.hpp"

namespace falcon_core::math::arrays {

LabelledControlArrays::LabelledControlArrays() = default;
LabelledControlArrays::LabelledControlArrays(
    const std::vector<LabelledControlArraySP>& arrays)
    : BaseLabelledArrays(arrays) {}
}  // namespace falcon_core::math::arrays

CEREAL_REGISTER_TYPE(falcon_core::math::arrays::LabelledControlArrays)
CEREAL_REGISTER_POLYMORPHIC_RELATION(
    falcon_core::math::arrays::BaseLabelledArrays<
        falcon_core::math::arrays::LabelledControlArray>,
    falcon_core::math::arrays::LabelledControlArrays)
